testsuite/classlibrary/TestString.sc

```diff
@@ -118,4 +118,17 @@ TestString : UnitTest {
 		this.assertEquals(result, expected);
 	}
 
+	test_findRegexp_nonEmptyResult {
+		var result = "the quick brown fox".findRegexp("[a-zA-Z]+").flop;
+		this.assert(
+			result[0] == [0, 4, 10, 16] and: { result[1][2] == "brown" },
+			"`\"the quick brown fox\".findRegexp(\"[a-zA-Z]+\")` should return 4 results at indices [0, 4, 10, 16], and the third word should be 'brown'"
+		)
```

Comment:
don't understand the need for the complexity of this test, it makes it difficult to understand and verify (why 4 matches, why only check the contents of the third match?). it could just as easily be `"a".findRegexp("a")`. maybe this function needs at most one more unit test to make sure it handles multiple matches, but beyond that it's basically testing boost.regex, which already has its own test suite :)

Reply:
OK, I've simplified it, though not quite as simple as you propose. My thinking is: We don't have a functional spec for these methods, and nobody has time to write one. (Documentation is no help here, either -- the sum total of the `findRegexp` documentation is "POSIX regular expression search" -- which is... that's a joke, right? Gosh, OK, let me add another commit here, before merging, to provide proper documentation.) So, if somebody does something to one of these methods in the future that breaks the output format, it would be helpful if the test suite includes an example of the result format (and not only for the most trivial case -- if the example result is only `[ [ 0, "a" ] ]`, it's hard to infer what 0 is... a character position? An index? but `[[0, "two"], [4, "words"]]` gives future developers more information).

Agreed that the test was too quirky before, no problem to change it.

Reply:
sure, i don't feel strongly about it. i would probably write a non-match, single match, and multiple match test to cover the sanity check cases.